### PR TITLE
Implemented front end tool transitions, front end tweaks (got rid of shape text), backend clean up, backend Shape abstract, line class, circle, rectangle, got rid of makeSelectable method

### PR DESCRIPTION
**Pull request by gnahzanit**
dont delete this branch I still need it this is just what I have so far

### CODE REVIEW

File: src/main/java/com/onionshop/controllers/ProjectStateController.java [hunk 1]
```diff
@@ -152,11 +144,12 @@ else if (colourButton.getButton() == MouseButton.SECONDARY) {
     protected void onCanvasMouseDragged(MouseEvent canvasMouseLocation) {
         int[][] updatedPixels = canvasInputProcessor.processControllerDataForDrawingManager(canvasMouseLocation);
 
-
         PixelWriter canvasPixelWriter = projectDrawing.getGraphicsContext2D().getPixelWriter();
         for (int i = 0; i < updatedPixels.length; i++) {
             canvasPixelWriter.setColor(updatedPixels[i][0], updatedPixels[i][1], currentCanvasColour);
         }
+
+        System.out.println(ToolStateManager.getInstance().getCurrentToolState());
```
Comment (faizahsayyid) [inline src/main/java/com/onionshop/controllers/ProjectStateController.java:152]: Is this necessary? Remember to get rid of this.

Comment (gnahzanit) [reply src/main/java/com/onionshop/controllers/ProjectStateController.java:152]: whoops no its not I tots forgot

File: src/main/java/com/onionshop/entities/Circle.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.onionshop.entities;
+
+public class Circle extends Shape implements Tool {
```
Comment (faizahsayyid) [inline src/main/java/com/onionshop/entities/Circle.java:3]: If shape implements tool, then Circle implements tool by default, you don't need to add this

Comment (gnahzanit) [reply src/main/java/com/onionshop/entities/Circle.java:3]: whoops forgot I think I added implements tool for circle & line & rectangle sorry bout that

File: src/main/java/com/onionshop/controllers/ProjectStateController.java [hunk 3]
```diff
@@ -168,35 +161,39 @@ protected void onToolSizeSliderMove() {
...
     /**
      * Takes in an event that changes the size of brush, update the brush size accordingly
      *
      */
     public void brushSizeUpdate(){
+
     }
 
+    @FXML
+    protected void onBrushPenClick() {
```
Comment (faizahsayyid) [inline src/main/java/com/onionshop/controllers/ProjectStateController.java:173]: add comment header

File: src/main/java/com/onionshop/controllers/ProjectStateController.java [hunk 4]
```diff
@@ -168,35 +161,39 @@ protected void onToolSizeSliderMove() {
...
-        brushPen.setText("Pen");
+    @FXML
+    public void onEraserClick() {
+        Eraser currentToolState = new Eraser("round", 10);
+        ToolStateManager.getInstance().setCurrentToolState(currentToolState);
     }
 
-    
 
-    public void onLineToolClick(ActionEvent actionEvent) {
```
Comment (faizahsayyid) [inline src/main/java/com/onionshop/controllers/ProjectStateController.java:194]: yeah just go back and add comment headers to all of these

Comment (gnahzanit) [reply src/main/java/com/onionshop/controllers/ProjectStateController.java:194]: will keep in mind for tmr's push ⭐️⭐️⭐️

**Comment by faizahsayyid** [comment]
Looks good, just make sure to add comment headers (javadoc)!!

**Pull request merged**